### PR TITLE
Enable Bind command on Open as Folder projects

### LINKED ISSUE #2562: Enable Bind command on Open as Folder projects
#### Description

Open as folder projects are not recognizable as bindable due to `KnownUIContexts.SolutionExistsAndFullyLoadedContext` and
`KnownUIContexts.SolutionExistsAndNotBuildingAndNotDebuggingContext` evaluating to `false` when in Open as Folder mode. See also #2555.

TODO:
- [x] Support showing the bind command when in Open as Folder mode

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Fixes #2562

### CODE REVIEW

File: src/Integration/Binding/BindingController.cs [hunk 1]
```diff
@@ -95,12 +99,13 @@ protected override int OnQueryStatus(ref Guid pguidCmdGroup, uint cCmds, OLECMD[
...
-                && VsShellUtils.IsSolutionExistsAndFullyLoaded()
-                && VsShellUtils.IsSolutionExistsAndNotBuildingAndNotDebugging()
-                && (this.projectSystemHelper.GetSolutionProjects()?.Any() ?? false);
+                   && args.ProjectKey != null
+                   && this.host.VisualStateManager.IsConnected
+                   && !this.host.VisualStateManager.IsBusy
+                   && (folderWorkspaceService.IsFolderWorkspace()
+                       || (VsShellUtils.IsSolutionExistsAndFullyLoaded()
+                           && VsShellUtils.IsSolutionExistsAndNotBuildingAndNotDebugging()
+                           && (this.projectSystemHelper.GetSolutionProjects()?.Any() ?? false)));
```
Comment (duncanp-sonar) [inline src/Integration/Binding/BindingController.cs:108]: `GetSolutionProjects` currently calls through to `ProjectToLanguageMapper.GetLanguageForProject`, so this condition will only be true if there is a bindable project.

We probably still want to check that condition for CMake projects, since it is a more strict check than just that the current solution is an "open as folder" one i.e.

```
...
&& (folderWorkspaceService.IsFolderWorkspace()
                       || (VsShellUtils.IsSolutionExistsAndFullyLoaded()
                           && VsShellUtils.IsSolutionExistsAndNotBuildingAndNotDebugging())
&& (this.projectSystemHelper.GetSolutionProjects()?.Any() ?? false)
```

File: src/Integration.UnitTests/Binding/BindingControllerTests.cs [hunk 2]
```diff
@@ -436,27 +486,33 @@ private static BindCommandArgs CreateBindingArguments(string key, string name, s
             return new BindCommandArgs(key, name, new ConnectionInformation(new Uri(serverUri)));
         }
 
-        private static ServerViewModel CreateServerViewModel()
```
Comment (duncanp-sonar) [inline src/Integration.UnitTests/Binding/BindingControllerTests.cs:439]: Presumably this just wasn't called?

File: src/Integration.UnitTests/Binding/BindingControllerTests.cs [hunk 3]
```diff
@@ -66,28 +68,32 @@ public class BindingControllerTests
         public void TestInitialize()
         {
             KnownUIContextsAccessor.Reset();
-            this.sonarQubeService = new Mock<ISonarQubeService>();
```
Comment (duncanp-sonar) [inline src/Integration.UnitTests/Binding/BindingControllerTests.cs:69]: There are lot of stylistic changes which make it difficult to focus on the real changes. Reviewing would be easier if they were done in separate commit. Also, something as straightforward as removing `this` everywhere doesn't really need to be reviewed, so it doesn't have to form part of the review.

**Review by duncanp-sonar — COMMENTED** [review]
Just a question about whether we should still check `projectSystemHelper.GetSolutionProjects` for "open as folder" projects.

File: src/Integration.UnitTests/Binding/BindingControllerTests.cs (hunk 3 above)
Comment (rita-sonarsource) [reply src/Integration.UnitTests/Binding/BindingControllerTests.cs:69]: You're right, my apologies.

File: src/Integration/Binding/BindingController.cs (hunk 1 above)
Comment (rita-sonarsource) [reply src/Integration/Binding/BindingController.cs:108]: Yes you're right, this would allow us to easy disable it for JS 👍
